src/atomic_c.c

```diff
@@ -118,72 +118,21 @@
 
 #endif /* ENABLE_PROFILING */
 
+#define SHMEM_DEF_SWAP(STYPE,TYPE,ITYPE) \
+    TYPE shmem_##STYPE##_swap(TYPE *target, TYPE value, int pe) \
+    {                                                           \
+        TYPE newval;                                            \
+        SHMEM_ERR_CHECK_INITIALIZED();                          \
+        shmem_internal_swap(target, &value, &newval,            \
+                            sizeof(TYPE), pe, ITYPE);           \
+        shmem_internal_get_wait();                              \
+        return newval;                                          \
+    }
 
-float
-shmem_float_swap(float *target, float value, int pe)
-{
-    float newval;
-
-    SHMEM_ERR_CHECK_INITIALIZED();
-
-    shmem_internal_swap(target, &value, &newval, sizeof(float), pe, SHM_INTERNAL_FLOAT);
-    shmem_internal_get_wait();
-    return newval;
-}
-
-
-double
-shmem_double_swap(double *target, double value, int pe)
-{
-    double newval;
-
-    SHMEM_ERR_CHECK_INITIALIZED();
-
-    shmem_internal_swap(target, &value, &newval, sizeof(double), pe, SHM_INTERNAL_DOUBLE);
-    shmem_internal_get_wait();
-    return newval;
-}
-
-
-int
-shmem_int_swap(int *target, int value, int pe)
-{
-    int newval;
-
-    SHMEM_ERR_CHECK_INITIALIZED();
-
-    shmem_internal_swap(target, &value, &newval, sizeof(int), pe, SHM_INTERNAL_INT);
-    shmem_internal_get_wait();
-    return newval;
-}
-
-
-long
-shmem_long_swap(long *target, long value, int pe)
-{
-    long newval;
-
-    SHMEM_ERR_CHECK_INITIALIZED();
-
-    shmem_internal_swap(target, &value, &newval, sizeof(long), pe, SHM_INTERNAL_LONG);
-    shmem_internal_get_wait();
-    return newval;
-}
-
-
-long long
-shmem_longlong_swap(long long *target, long long value, int pe)
-{
-    long long newval;
-
-    SHMEM_ERR_CHECK_INITIALIZED();
-
-    shmem_internal_swap(target, &value, &newval, sizeof(long long), pe, SHM_INTERNAL_LONG_LONG);
-    shmem_internal_get_wait();
-    return newval;
-}
+SHMEM_EVAL_MACRO_FOR_EXTENDED_AMO_WITH_ITYPE(SHMEM_DEF_SWAP,)
```

Comment:
I'm probably missing something but doesn't this macro only include float/double? what about long/int/longlong